### PR TITLE
feat: turn onConferenceInfoChange async WBP-5722

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Turn `generateConferenceInfo` async and `onConferenceInfoChange ` into an async sequence. This in preparation for upgrading CC which will force this change.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-data-model/Source/MLS/MLSService.swift [hunk 1]
```diff
@@ -343,15 +343,21 @@ public final class MLSService: MLSServiceInterface {
...
-            $0.isOne(of: parentGroupID, subConversationGroupID)
-        }.compactMap { [weak self] _ in
-            try? self?.generateConferenceInfo(
-                parentGroupID: parentGroupID,
-                subconversationGroupID: subConversationGroupID
-            )
-        }.eraseToAnyPublisher()
+    ) -> AsyncStream<MLSConferenceInfo> {
+        var sequence = onEpochChanged()
+            .buffer(size: 1000, prefetch: .keepFull, whenFull: .dropOldest)
```
Comment (netbe) [inline wire-ios-data-model/Source/MLS/MLSService.swift]: use constant, I wonder why 1000, any reason?

File: wire-ios-data-model/Source/MLS/MLSService.swift [hunk 2]
```diff
@@ -343,15 +343,21 @@ public final class MLSService: MLSServiceInterface {
...
+            .filter({ $0.isOne(of: parentGroupID, subConversationGroupID) })
+            .values
+            .compactMap({ [weak self] _ in
+                try? await self?.generateConferenceInfo(
```
Comment (netbe) [inline wire-ios-data-model/Source/MLS/MLSService.swift]: does the error needs to be propagated?

File: wire-ios-data-model/Source/MLS/MLSService.swift (hunk 1 above)
Comment (typfel) [reply wire-ios-data-model/Source/MLS/MLSService.swift]: No good reason, needs be higher than the number of epoch change we could potentially receive before we start consuming the sequence.

I can introduce a constant.

File: wire-ios-data-model/Source/MLS/MLSService.swift (hunk 2 above)
Comment (typfel) [reply wire-ios-data-model/Source/MLS/MLSService.swift]: We probably should, I just kept the way it was.

Comment (typfel) [reply wire-ios-data-model/Source/MLS/MLSService.swift]: Updated to propagate the error

**Review by netbe — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]

**Review by caldrian — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]

**Pull request merged**